File: src/main/java/cherkasov/com/streamsource/FakeConnection.java
package cherkasov.com.streamsource;

import java.io.IOException;
import java.io.InputStream;

/**
 * Connection for input stream from memory buffer
 */
public class FakeConnection implements Connection {
    private InputStream inputStream;
    private boolean connected;

    public FakeConnection(InputStream inputStream) {

        this.inputStream = inputStream;
        this.connected = true;
    }

    @Override
    public boolean connect() {

        return true;
    }

    @Override
    public boolean isConnected() {

        return connected;
    }


    @Override
    public long getContentLength() throws IOException {

        return inputStream.available();
    }

    @Override
    public InputStream getInputStream() {

        return inputStream;
    }

    @Override
    public void disconnect() throws IOException {

            inputStream.close();
    }
}

File: src/main/java/cherkasov/com/streamsource/FakeConnection.kt
package cherkasov.com.streamsource

import java.io.IOException
import java.io.InputStream

/**
 * Connection for input stream from memory buffer
 */
class FakeConnection(override val inputStream: InputStream) : Connection {
    override val isConnected = true
    override fun connect(): Boolean {
        return true
    }

    @get:Throws(IOException::class)
    override val contentLength: Long
        get() = inputStream.available().toLong()

    @Throws(IOException::class)
    override fun disconnect() {
        inputStream.close()
    }

}